### PR TITLE
Add Search Endpoint

### LINKED ISSUE #2: Server returns a limited list of items for a given Geolocation

### PR DESCRIPTION
**Pull request by rluetzner**
This adds a search endpoint to our backend API.

The endpoint is reachable via `/api/search` and requieres two querey parameters:
- `topLeftLocation`: this is a comma separated value of latitude and longitued (in this order) and defines the upper left corner of a search window.
- `lowerRightLocation`: this is a comma separated value of latitude and longitued (in this order) and defines the lower right corner of a search window.

The response is an list of items that are in the given search window.

An example request would be `/api/search?topLeftLocation=52.5,13&lowerRightLocation=54,14`.

This fixes #2 .

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,9 +2,40 @@ const express = require('express')
...
+  tlSplit = req.query.topLeftLocation ?
+    req.query.topLeftLocation.split(",")
+    : [];
+  lrSplit = req.query.lowerRightLocation ?
+    req.query.lowerRightLocation.split(",")
+    : [];
+
+  if (tlSplit.length != 2 || lrSplit.length != 2) {
+    res.status(400).send({
+      message: 'Invalid location data.'
```
Comment (aweisser) [inline index.js]: Perhaps add some information how valid location data should look like

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            lat: 54,
+            lng: 14
+        };
+        items = mockDb.itemsByLocation(tlLocation, lrLocation)
+        expect(items.length).to.be.equal(mockdb.availableItems.length);
+    })
+})
+
+describe('itemsByLocation(...)', function () {
+    it('should return no items in an invalid window', function () {
```
Comment (aweisser) [inline test/test.js]: Is this  an invalid window or a valid window without items?

File: mockdb.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+module.exports = {
+    availableItems: [
+        {
+            user: "N. Maier",
```
Comment (aweisser) [inline mockdb.js]: I would suggest Fantasy Usernames like "Morpheus" or "Gandalf" to underline the anonymity of the concept.

File: index.js (hunk 1 above)
Comment (rluetzner) [reply index.js]: I resolved this in #21 .

File: mockdb.js (hunk 3 above)
Comment (rluetzner) [reply mockdb.js]: Sounds good to me. Anonymity is a good point, I didn't think of that.

Comment (rluetzner) [reply mockdb.js]: I fixed this in #21 .

File: test/test.js (hunk 2 above)
Comment (rluetzner) [reply test/test.js]: I added validation to the request handler function. This makes more sense in the long term, as the MockDd class is only useful as long as we don't have an actual database.

**Pull request merged**